### ISSUE TITLE
Python lib examples only working with "with"

### ISSUE THREAD

**Issue by SebastianSchildt** [eclipse/kuksa.val#476]
Current python lib uses "context managers" to enable usage in `with` statements. The canonical example from the documentation is

```python
with VSSClient('127.0.0.1', 55556) as client:
    current_values = client.get_current_values([
        'Vehicle.Speed',
    ])
    print(current_values)
```

the following code does not work

```python
databroker=VSSClient('127.0.0.1', 55556)
current_values2 = databroker.get_current_values([
    'Vehicle.Speed',
])
print(current_values2)
```

this is unexpected and not how it works for example with file io in python. here

```python
file = open("some.file", "r")
```

can achieve the same as

```python
with open("some.file", "r") as file:
```

The root cause seems to be, that without `with``__enter__`is not called, see the error message for the second example

```
Traceback (most recent call last):  File "/Users/scs2rng/Documents/Dev/kuksa-prometheus/test.py", line 11, in <module>    current_values2 = databroker.get_current_values([  File "/Users/scs2rng/.local/share/virtualenvs/kuksa-prometheus-n8PS3LmC/lib/python3.10/site-packages/kuksa_client/grpc/__init__.py", line 586, in get_current_values    entries = self.get(  File "/Users/scs2rng/.local/share/virtualenvs/kuksa-prometheus-n8PS3LmC/lib/python3.10/site-packages/kuksa_client/grpc/__init__.py", line 738, in get    resp = self.client_stub.Get(req, **rpc_kwargs)AttributeError: 'NoneType' object has no attribute 'Get'
```

It seems a bit unclear what the best python (coding) practice to achieve a better behavior is

**Comment by erikbosch** [comment]
If we change something we must make sure feeders, still work. As an example dbcfeeder creates client with:

```
 vss_client = self._exit_stack.enter_context(kuksa_client.grpc.VSSClient(
                host=self._kuksa_client_config['ip'],
                port=self._kuksa_client_config['port'],
            ))
```

(Which results in that `__enter__`method is called)

**Comment by SebastianSchildt** [comment]
Hah, that is likely the "workaround" if you want a vss_client object. 
I have seen code where `__enter__` basically just returns ´self´, and the initialisation/acquiring of resources is already done in `__init__`. Not sure that is how context managers are intended to be used, but it still gives the advantage of cleanup via `__exit__`  should something go wrong, and would probably be compatible with existing code.

So that seems like a possible solution to me, that would make it easier/more obvious to use for the average py rookie. Not sure it is clean though.

Maybe @romainletendart  is still around having an opinion?

**Comment by romainletendart** [comment]
Hi guys!

Yes, in its current state, `VSSClient` is only meant to be used as a context manager.

In order to get it working the way you want, you could move logic from `__enter__` to a new `connect` method (respectively moving logic from `__exit__` to a new `disconnect` method).
Then you could call `connect` from within `__enter__` to keep on supporting the context manager usage (respectively `disconnect` from within `__exit__`).

However, I wouldn't recommend calling `connect` from within `__init__` as one should be able to create a client object without initiating I/O. However you could check from within `get`, `set`, `subscribe` methods if the client is connected and connect it if it isn't.

**Comment by SebastianSchildt** [comment]
fixed in #526
